tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -0,0 +1,98 @@
+"""Tests for RH Cloud - Inventory
+
+:Requirement: RH Cloud - Inventory
+
+:CaseAutomation: Automated
+
+:CaseLevel: System
+
+:CaseComponent: RHCloud-Inventory
+
+:Assignee: jpathan
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+from datetime import datetime
+from datetime import timedelta
+
+import pytest
+
+from robottelo.api.utils import wait_for_tasks
+from robottelo.config import settings
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.tier3
+def test_rhcloud_insights_e2e(
+    session, rhel8_insights_vm, fixable_rhel8_vm, organization_ak_setup, enable_lab_features
+):
+    """Synchronize hits data from cloud, verify it is displayed in Satellite and run remediation.
+
+    :id: d952e83c-3faf-4299-a048-2eb6ccb8c9c2
+
+    :Steps:
+        1. Prepare misconfigured machine and upload its data to Insights.
+        2. Add Cloud API key in Satellite.
+        3. In Satellite UI, Configure -> Insights -> Add RH Cloud token and syns recommendations.
+        4. Run remediation for dnf.conf recommendation against rhel8 host.
+        5. Assert that job completed successfully.
+        6. Sync Insights recommendations.
+        7. Search for previously remediated issue.
+
+    :expectedresults:
+        1. Insights recommendation related to dnf.conf issue is listed for misconfigured machine.
+        2. Remediation job finished successfully.
+        3. Insights recommendation related to dnf.conf issue is not listed.
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    query = 'dnf.conf'
+    job_query = (
+        f'Remote action: Insights remediations for selected issues on {rhel8_insights_vm.hostname}'
+    )
+    with session:
+        session.organization.select(org_name=org.name)
+        session.cloudinsights.save_token_sync_hits(settings.rh_cloud.token)
+        # This is a workaround for BZ#1983575. Remove following two code lines once BZ is fixed.
+        session.browser.refresh()
+        session.cloudinsights.sync_hits()
+
+        timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime('%Y-%m-%d %H:%M')
```

Comment:
Is it really necessary to wait 2 minutes to check the job has started 2 minutes ago? I would prefer to check for some job state. Was this discussed with the developers?

Reply:
@jaryn nice catch. It's not required I had added it by mistake. Updated the pr.